### ISSUE TITLE
Could not find dependency Class[Apt::Update] for Package[php7.0-cli]

### ISSUE THREAD

**Issue by mwhahaha** [voxpupuli/puppet-php#227]
It appears that there are multiple places where packages have dependencies on Class['apt::update'] but no actual inclusion of this class. There should be an include ::apt somewhere or if this needs to be done in the user's role/profile definitions then it should to be documented.

```
ubuntu:~/aws$ cat test.pp
class { '::php':
  manage_repos => false,
}
ubuntu:~/aws$ puppet apply --modulepath `pwd`/modules test.pp
Warning: Config file /home/ubuntu/.puppet/hiera.yaml not found, using Hiera defaults
Notice: Compiled catalog for myhost.com in environment production in 0.47 seconds
Error: Could not find dependency Class[Apt::Update] for Package[php7.0-cli] at /home/ubuntu/aws/modules/php/manifests/packages.pp:36
ubuntu:~/aws$ puppet module list --modulepath `pwd`/modules
/home/ubuntu/aws/modules
├── darin-zypprepo (v1.0.2)
├── example42-puppi (v2.1.13)
├── example42-yum (v2.1.27)
├── mayflower-php (v4.0.0-beta1)
├── nanliu-staging (v1.0.3)
├── puppetlabs-apache (v1.9.0)
├── puppetlabs-apt (v2.2.2)
├── puppetlabs-concat (v2.1.0)
├── puppetlabs-inifile (v1.5.0)
├── puppetlabs-mysql (v3.6.2)
└── puppetlabs-stdlib (v4.11.0)
```

**Comment by mmoll** [comment]
related to 7967a64

**Comment by cliff-wakefield** [comment]
Running 4.0.0-beta1 and still seem to be hitting this on Ubuntu 16.04?

I looked at the changes that tdm4 suggested and they look like they have been incorporated but still having issues?

**Comment by mmoll** [comment]
While fixing another bug, that change introduced this exact problem.

**Comment by roll4life** [comment]
forgive my lack of puppet knowledge.
would it be a reasonable solution to add the include to the top of the packages 
`class php::packages (
    include '::apt'
. . . `

**Comment by mmoll** [comment]
something like using `Class['::php::repo']`, but with owrking anchoring would be the solution

**Comment by roll4life** [comment]
so something like this in the init.pp
```
  anchor { 'php::begin': } ->
    class { '::apt': } ->
    class { '::php::packages': } ->
    class { '::php::cli':
      settings => $real_settings,
    } ->
  anchor { 'php::end': }
```
I can only say that it does work in ubuntu 16.04 with `manage_repos => false`

**Comment by cliff-wakefield** [comment]
As a work around I have simply done the following to allow my project to continue.

`php/manifests/packages.pp`
```
  if $::operatingsystem == 'Ubuntu' {
    include ::apt
    package { $real_names:
      ensure  => $ensure,
      require => Class['::apt::update'],
    }
  } else {
    package { $real_names:
      ensure => $ensure,
    }
  }
```

I don't know if other operating systems need the include as well.  If so it can be moved outside of the if block.

**Comment by roll4life** [comment]
or you could use 
`if $osfamily == 'debian'`
which would include all debian derivatives.

**Comment by cliff-wakefield** [comment]
Any update on when this maybe resolved?

**Comment by roll4life** [comment]
I'll go ahead and create the PR with my previous proposal, if it is good hopefully someone will merge it.

**Comment by cliff-wakefield** [comment]
My pull request has now been merged which solves the issue on Debian based distributions.

**Comment by mbrodala** [comment]
For the reference: #250

**Comment by juniorsysadmin** [comment]
Appears to be fixed by #250 , closing

**Comment by KlavsKlavsen** [comment]
Actually that patch does NOT resolve it, if you use manage_repos => false (which I need to since my hosts have no outbound internetaccess so its mirrors all the way :)
